Sources/GRPC/SpanStatus+GRPCStatus.swift

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import TracingInstrumentation
+
+extension SpanStatus {
+  /// Initialize a `SpanStatus` by mapping the given `GRPCStatus`.
+  /// - Parameter status: The `GRPCStatus` to map to a `SpanStatus`.
+  init(_ status: GRPCStatus) {
+    let canonicalCode: SpanStatus.CanonicalCode
+
+    switch status.code {
+    case .ok:
+      canonicalCode = .ok
+    case .cancelled:
+      canonicalCode = .cancelled
+    case .unknown:
+      canonicalCode = .unknown
+    case .invalidArgument:
+      canonicalCode = .invalidArgument
+    case .deadlineExceeded:
+      canonicalCode = .deadlineExceeded
+    case .notFound:
+      canonicalCode = .notFound
+    case .alreadyExists:
+      canonicalCode = .alreadyExists
+    case .permissionDenied:
+      canonicalCode = .permissionDenied
+    case .resourceExhausted:
+      canonicalCode = .resourceExhausted
+    case .failedPrecondition:
+      canonicalCode = .failedPrecondition
+    case .aborted:
+      canonicalCode = .aborted
+    case .outOfRange:
+      canonicalCode = .outOfRange
+    case .unimplemented:
+      canonicalCode = .unimplemented
+    case .internalError:
+      canonicalCode = .internal
+    case .unavailable:
+      canonicalCode = .unavailable
+    case .dataLoss:
+      canonicalCode = .dataLoss
+    case .unauthenticated:
+      canonicalCode = .unauthenticated
+    case .doNotUse:
```

Comment:
FYI: I just filed #942 -- `doNotUse` made it's way over from the old gRPC Swift implementation which in turn came from [here](https://github.com/grpc/grpc/blob/c79bef55ee348c298cbdcc60448d8e70e4972a19/include/grpcpp/impl/codegen/status_code_enum.h#L138). Its purpose was to force a `default` branch, which we should do here.

Reply:
Cool 👍 Updated it to use `default` instead.

Reply:
This is a great start by the way; please @ me if there's anything you want me to look at specifically :)

Reply:
Thanks, I appreciate that! 😎